lib/rmt/scc.rb

```diff
@@ -86,16 +86,21 @@ def sync_systems
     # do not sync BYOS proxy systems to SCC
     systems = System.where('scc_registered_at IS NULL OR last_seen_at > scc_registered_at').where(proxy_byos: false)
     @logger.info(_('Syncing %{count} updated system(s) to SCC') % { count: systems.size })
-    scc_api_client.send_bulk_system_update(systems) do |successful_response|
-      next if successful_response[:systems].count == 0
-
-      successful_response[:systems].each do |system_hash|
-        # Update attributes without triggering after_update callback (which resets scc_synced_at to nil)
-        System.find_by(login: system_hash[:login])
-           .update_columns(scc_system_id: system_hash[:id], scc_synced_at: Time.current)
-      rescue StandardError => e
-        @logger.error(_('Failed to sync systems: %{error}') % { error: e.to_s })
-      end
+
+    updated_systems = scc_api_client.send_bulk_system_update(systems)
+    if updated_systems[:systems].blank?
+      # Case: silently fail but respond with 201
+      systems.update_all(scc_registered_at: Time.current)
+      @logger.info(_('Updated system sync date'))
+      return
+    end
+
+    updated_systems[:systems].each do |system_hash|
+      # Update attributes without triggering after_update callback (which resets scc_synced_at to nil)
+      System.find_by(login: system_hash[:login], password: system_hash[:password], system_token: system_hash[:token])
```

Comment:
Wouldn't the `system_hash[:token]` be the system id in that case and not the token? 
See https://github.com/SUSE/rmt/blob/6da777085d3e11b287c4ca4c04c52fc0ace65625/lib/suse/connect/api.rb#L106

So I think this might need to be: 

```suggestion
      if system_hash[:token]
        System.find_by(login: system_hash[:login], id: system_hash[:token])
      else
         System.find_by(login: system_hash[:login]) 
      end
```

Reply:
I think I will revert this change. The `id` isn't token but the `scc_system_id`.

Reply:
It's a bit weird, but the value we receive here from SCC as `system_hash[:token]` is actually not the `system_token` RMT has stored for the system, but it's `id`.
That was done to have a static token in the sync with SCC to identify the system to be updated on SCC side. There's a comment describing that here: https://github.com/SUSE/rmt/blob/6da777085d3e11b287c4ca4c04c52fc0ace65625/lib/suse/connect/api.rb#L100